cmd/cluster/encryption/encryption.go

```diff
@@ -70,6 +70,50 @@ var listCmk = &cobra.Command{
 	},
 }
 
+var updateCmkState = &cobra.Command{
+	Use:   "update-state",
+	Short: "Update Encryption at Rest (EaR) state for a cluster",
+	Long:  "Update Encryption at Rest (EaR) state for a cluster",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("Could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+		clusterName, _ := cmd.Flags().GetString("cluster-name")
+		clusterId, err := authApi.GetClusterIdByName(clusterName)
+		if err != nil {
+			logrus.Fatalf("%s", ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		resp, r, err := authApi.ListClusterCMKs(clusterId).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		if resp.Data == (ybmclient.NullableCMKSpec{}) {
+			logrus.Fatalf("No Encryption at rest configuration found for this cluster")
```

Comment:
btw why do we need to check this? I looked at the PR we seem to already throw an error if we perform this operation for a CMK unconfigured cluster

Reply:
We do that for every command

Reply:
This is specific to this CMK command

Reply:
Actually, the get cluster cmk api does not throw an error if the CMK config is not found, it simply returns null.
So it needs to be covered on the cli side.

Reply:
Do we need to check this at the CLI side at all. If you submit UpdateCMKStateSpec API it will also check for the same and throw an appropriate error. So do we need to check this at CLI?

Reply:
The update cluster cmk api needs cmkId as a path parameter, so if I call this API, it is going to throw 404 error, because the path will be changed now.
So it needs to be handled on the CLI side and the API to be called only when the EAR configuration is present for the cluster.

Reply:
We need the API call to get the cmk id anyway.

Reply:
@kc-yb I don't think any additional API calls are being made, the nil check will throw an error before calling the update API if needed.